elements/pfe-tooltip/BaseTooltip.ts

```diff
@@ -0,0 +1,92 @@
+import { LitElement, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { Placement } from '@popperjs/core';
+import { getRandomId } from '@patternfly/pfe-core/functions/random.js';
+import { FloatingDOMController } from '@patternfly/pfe-core/controllers/floating-dom-controller.js';
+
+export abstract class BaseTooltip extends LitElement {
+  @property({ reflect: true, type: String }) position = 'top';
+
+  @property({ reflect: true, type: Array }) offset = [0, 15];
+
+  #id = getRandomId(this.localName);
+
+  #domController: FloatingDOMController = new FloatingDOMController(this);
+
+  get #isOpen(): boolean {
+    return this.#domController.getOpen();
+  }
+
+  get #invoker() {
+    return this.shadowRoot?.querySelector('#invoker');
+  }
+
+  get #tooltip() {
+    return this.shadowRoot?.querySelector(`[id^=${this.localName}`);
+  }
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+    if (!['top', 'bottom'].includes(this.position)) {
+      this.offset = [-4, 15];
+    }
+    this._addListeners();
+  }
+
+  override disconnectedCallback() {
+    super.disconnectedCallback();
+    this._removeListeners();
+  }
+
+  override firstUpdated(): void {
+    if (this.#invoker && this.#tooltip) {
+      this.#domController.create(this.#invoker, this.#tooltip as HTMLElement, this.position as Placement, this.offset);
```

Comment:
Consider changing the position type of `String` to the popper `Placement` type and drop the `as`

```Suggestion
  @property({ reflect: true }) position: Placement = 'top';

  @property({ reflect: true, type: Array }) offset = [0, 15];

  #id = getRandomId(this.localName);

  #domController: FloatingDOMController = new FloatingDOMController(this);

  get #isOpen(): boolean {
    return this.#domController.getOpen();
  }

  get #invoker() {
    return this.shadowRoot?.querySelector('#invoker');
  }

  get #tooltip() {
    return this.shadowRoot?.querySelector(`[id^=${this.localName}`);
  }

  override connectedCallback(): void {
    super.connectedCallback();
    if (!['top', 'bottom'].includes(this.position)) {
      this.offset = [-4, 15];
    }
    this._addListeners();
  }

  override disconnectedCallback() {
    super.disconnectedCallback();
    this._removeListeners();
  }

  override firstUpdated(): void {
    if (this.#invoker && this.#tooltip) {
      this.#domController.create(this.#invoker, this.#tooltip as HTMLElement, this.position, this.offset);
```